content/BatchRequestContentCollection.md

```diff
@@ -0,0 +1,23 @@
+# BatchRequestContentCollection
+
+## Objectives
+
+Provide a class that does not have limitation of that 20 requests by creating new `BatchRequestContent` objects to hold new requests to simplify the batching for advanced scenarios. 
+The `BatchRequestContentCollection` is essentially a collection wrapper for `BatchRequestContent` and is not expected to be serializable as the individual `BatchRequestContent` instance are sent out independently and will receive independent responses.
+
+### Requirements
+
+Similar to [BatchRequestContent](BatchRequestContent.md), it must support the below functionality while holding a collection of BatchRequestContent instances.
+
+- It MUST be possible to add native platform HTTP request objects to a BatchRequestContent object. The request URL MUST be a relative URL. Adding Graph specific Request objects MAY be supported.
+- Added requests MUST be assigned unique identifiers. Those identifiers MAY be user provided or autogenerated. The uniqueness SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in AND across the `BatchRequestContent` instances.
+- It MUST be possible to create dependencies between request objects. However, these dependencies SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in and not across `BatchRequestContent` instances.
```

Comment:
have we considered the dependencies scenario? i.e. if request 21 depends on request 1, how do we make sure those are sequenced properly? or added to the same batch?

Reply:
Was going to ask the same. Or is it that the batch should automatically manage its size based on the dependencies and throw if we can't fit a set of dependencies within a single batch?

Reply:
The idea is as @sebastienlevert says, the addition of new requests with the dependsOn should be validating on addition and throw/not add the message as this will fail the constraint. Will update the wording here to capture this.

Reply:
Added point to cover this.